### PR TITLE
MNT/ENH: Add FLOAT and FILL options for spreadsheet dataType

### LINKED ISSUE #697: BUG - FLOAT and FILL data types not supported in XTCE converter
### Description of the issue

Some spreadsheets have FILL or FLOAT in the `dataType` column.  Currently, these just get skipped so there is no `ParameterTypeRef` in the XTCE document. We should warn if something isn't valid, and add these two types to the options.

### Steps to reproduce the issue

Change a dataType field to FLOAT, and see that it isn't output in the parameterTypeRef list.

### Expected behavior (What should happen)

_No response_

### Actual behavior (What does happen)

_No response_

### Code Snippet:

```python
Code
```

### Additional notes, affected areas, and suggested fixes

_No response_

### PR DESCRIPTION
**Pull request by greglucas**
# Change Summary

## Overview

This adds two new options to the handled data types, and also raises an error if we get a datatype that isn't handled. This is important because otherwise there will be a parameter added without a corresponding parameterType entry.

Working from an SCID spreadsheet, I now get the following parameterTypes, when they were missing before.

```
			<xtce:IntegerParameterType name="SCID_X252_FILL6" signed="false">
				<xtce:IntegerDataEncoding sizeInBits="6" encoding="unsigned" />
			</xtce:IntegerParameterType>
			<xtce:FloatParameterType name="SCID_X252_SCIENCEDATA1HZ_QUAT_10_HZ_TIME">
				<xtce:FloatDataEncoding sizeInBits="64" encoding="IEEE-754" />
			</xtce:FloatParameterType>
```

closes #697

### CODE REVIEW

File: imap_processing/tests/ccsds/test_data/excel_to_xtce_test_file.xlsx
Comment (greglucas) [inline imap_processing/tests/ccsds/test_data/excel_to_xtce_test_file.xlsx:1]: I went down the road of mocking/creating this spreadsheet in code, but it ended up being a lot more confusing in my opinion to follow what was being changed/updated. So this seemed reasonable to add a small-ish binary file for testing this routine for clarity's sake.

**Comment by greglucas** [comment]
@tech3371, ping for review on this again.

File: imap_processing/tests/ccsds/test_excel_to_xtce.py [hunk 1]
```diff
@@ -12,21 +12,36 @@
...
-    p.touch()
+def excel_file():
+    p = Path(__file__).parent / "test_data" / "excel_to_xtce_test_file.xlsx"
     return p
 
 
+def test_generated_xml(excel_file, tmp_path):
+    """Make sure we are producing the expected contents within the XML file.
+
+    To produce a new expected output file the following command can be used.
```
Comment (tech3371) [inline imap_processing/tests/ccsds/test_excel_to_xtce.py:23]: Good to know! This makes me wonder, do we have readme for this cli tool?

**Review by tech3371 — APPROVED** [review]
LGTM!

**Review by tmplummer — APPROVED** [review]
LGTM. I will admit that I am not up to speed on the xtce requirements though.

File: imap_processing/tests/ccsds/test_excel_to_xtce.py (hunk 1 above)
Comment (greglucas) [reply imap_processing/tests/ccsds/test_excel_to_xtce.py:23]: We have an issue to make this better! On my list...
https://github.com/IMAP-Science-Operations-Center/imap_processing/issues/703

**Pull request merged**